lib/trie/database.go

```diff
@@ -23,6 +23,21 @@ var ErrEmptyProof = errors.New("proof slice empty")
 // and the value is the encoded node.
 // Generally, this will only be used for the genesis trie.
 func (t *Trie) Store(db chaindb.Database) error {
+	for _, v := range t.childTries {
+		if v == nil {
+			// there was a child trie with this hash earlier, but
+			// it has changed since then. When storing child trie at
+			// a new hash, we store at the previous hash.
+			// TODO: delete the hash entry instead of setting it to nil?
+			continue
```

Comment:
this shouldn't happen, the entry should be deleted from the map

Reply:
addressed